### DISCUSSION TITLE
Can DuckDB pushdown list_cosine_similarity and list_distance functions to LanceDB?

### DISCUSSION THREAD

**Discussion by dmackenthun** [lance-format/lance#2020] [General]
Hi all,

Are there any vector search functions like _list_cosine_similarity_ and _list_distance_ that can be pushed down from DuckDB into LanceDB? I have a use case where I would like to do a similarity search within the context of a DuckDB SQL query. LanceDB is an intriguing option for persisting embeddings and performing semantic search, however, I need to be able to execute complex joins, nested queries, and filters within the same query.

Is there a different way to do this type of search in a DuckDB query that references a LanceDB table?

**Comment by wjones127** [comment]
We don't support pushdown like that, but you could define a [Python UDF](https://duckdb.org/docs/api/python/function.html) in DuckDB and do the computations in DuckDB.
